src/pages/Market/components/AllItem.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState,useEffect, useCallback } from 'react';
+import SearchIcon from '../../../assets/images/searchicon.png';
+import { Link } from 'react-router-dom';
+import ItemCard from './ItemCard';
+import { getProducts } from '../../../api/itemApi';
+
+const getPageSize = () => {
+    const width = window.innerWidth;
+    if(width < 767) {
+        return 4;
+    } else if (width < 1280) {
+        return 6;
+    } else {
+        return 10;
+    }
+};
+
+function AllItem() {
+    const [itemList, setItemList] = useState([]);
+    const [pageSize, setPageSize] = useState(getPageSize());
+    const [order, setOrder] = useState('recent');
+    const [selected, setSelected] = useState('최신순');
+    const [dropdownOpen, setDropdownOpen] = useState(false);
```

Comment:
P2:
타입을 추측할 수 있는 이름을 추천드립니다~

```suggestion
    const [isDropdownOpen, setIsDropdownOpen] = useState(false);
```